cmd/adsysd/client/service.go

```diff
@@ -115,9 +115,9 @@ func (a *App) serviceStop(force bool) error {
 		return err
 	}
 
-	if _, err := stream.Recv(); err != nil && !errors.Is(err, io.EOF) {
+	if _, err := stream.Recv(); err != nil {
 		// Ignore "transport is closing" error if force (i.e immediately drop all connections) was used.
-		if force && status.Code(err) == codes.Unavailable {
+		if (force && status.Code(err) == codes.Unavailable) || errors.Is(err, io.EOF) {
```

Comment:
I find that there is no explanation for this change. Why was it sometimes covered and not other times? Was it the test setup being flaky, should we rather fix the test?

Reply:
It would definitely be better to fix the test. But, if I understood correctly, the inside `if` was a "patch" to handle a status code that sometimes is emitted by grpc if we close all connections to quickly. 
I tried increasing the sleep time in the tests and some other things, but this instability isn't our fault.
Also, it's not really that wrong, is it? Since we are evaluating the error anyway.

Reply:
> It would definitely be better to fix the test. But, if I understood correctly, the inside if was a "patch" to handle a status code that sometimes is emitted by grpc if we close all connections to quickly.

That’s why I think we should rather work on the test to ensure we are doing what we expect (not closing the connection before some data is available for instance). Because, here, the workaround will make that sometimes we should exit early when we don’t. So, yeah, the coverage itself is fixed, but the test is not testing what it’s supposed to test :)

Reply:
The only way for us to consistently cover this line is to be able to consistently interrupt the cat hanging call and we can't really do that, or can we? (This is a real question). We can't know for sure the execution time of it, so the value will keep floating around and so will the coverage.

Reply:
Nothing else than what was stated above IIRC?

Reply:
> Referring to this question specifically[…]

Yes, there are 2 ways to close it:
- send SIGTERM (but ofc, we can’t do that in our poor test process)
- the client gets `io.EOF`, but this is service-side generated.

I think a good comment of what happens and why we need to move this (where is the race exactly in the tests that makes the line cover/not cover) will be enough then.